### PR TITLE
Add babel-lodash plugin

### PR DESCRIPTION
**Pull request by meckila-navapbc**
### Description
Add babel lodash plugin to transform full imports of lodash and reduce bundle size
[HackMD](https://hackmd.io/MQIKaQItQ0WoMNgOa7LxHA?both)

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] Tests pass

### CODE REVIEW

**Review by allthesignals — APPROVED** [review]
Looks great! Sorry mostly just have questions... was the main purpose here to satisfy babel-plugin-lodash's requirement that method chaining needs to be refactored into nested methods so that the plugin can actually tree-shake? Were there any codemods for this sort of thing? 

Do you know much the bundle shrunk thanks to tree-shaking? Looking at your webpack-bundle-analyzer diagram, is that what is actually shipped to the users machine or is it just the development environment (asking because there's Sass present in the bundle).

As for the change from method-chaining to nesting, maybe someday in the future we'll get to use some kind of [pipe operator](https://github.com/tc39/proposal-pipeline-operator) :)

I've enjoyed looking through this and excited to learn more from you! Thanks!

File: client/app/queue/components/IssueRemandReasonsOptions.jsx [hunk 1]
```diff
@@ -153,38 +164,50 @@ class IssueRemandReasonsOptions extends React.PureComponent {
...
       <React.Fragment key={option.id}>
-        <Checkbox name={rowOptId} onChange={onChange} value={values[option.id].checked} label={option.label} unpadded />
-        {values[option.id].checked && (
+        <Checkbox
+          name={rowOptId}
+          onChange={onChange}
+          value={checkboxValues[option.id].checked}
+          label={option.label}
+          unpadded
+        />
```
[the remark this answers is not in the document]
Comment (meckila-navapbc) [reply client/app/queue/components/IssueRemandReasonsOptions.jsx:180]: Partially a lint fix and partially to clarify the variable name `values` to `checkboxValues`

File: client/app/reader/DocumentCategoryIcons.jsx [hunk 2]
```diff
@@ -1,45 +1,57 @@
...
+  sortBy(
+    Object.keys(Constants.documentCategories).reduce((list, name) => {
+      if (document[categoryFieldNameOfCategoryName(name)]) {
+        return {
+          ...list,
+          [name]: Constants.documentCategories[name]
+        };
+      }
+
+      return list;
```
[the remark this answers is not in the document]
Comment (meckila-navapbc) [reply client/app/reader/DocumentCategoryIcons.jsx:20]: To work around the chaining that the plugin doesn't allow, the reducer is replacing the lodash `filter` function specifically.

File: client/app/hearings/utils.js [hunk 3]
```diff
@@ -59,25 +74,22 @@ export const sortHearings = (hearings) =>
   );
 
 export const filterIssuesOnAppeal = (issues, appealId) =>
-  _(issues).
-    omitBy('_destroy').
-    pickBy({ appeal_id: appealId }).
-    value();
+  pickBy(omitBy(issues, '_destroy'), { appeal_id: appealId });
```
[the remark this answers is not in the document]
Comment (meckila-navapbc) [reply client/app/hearings/utils.js:77]: Not to my knowledge! lodash needs `.value` to execute chaining sequences because they are lazily evaluated

**Comment by meckila-navapbc** [comment]
> Looks great! Sorry mostly just have questions... was the main purpose here to satisfy babel-plugin-lodash's requirement that method chaining needs to be refactored into nested methods so that the plugin can actually tree-shake? Were there any codemods for this sort of thing?
> 
> Do you know much the bundle shrunk thanks to tree-shaking? Looking at your webpack-bundle-analyzer diagram, is that what is actually shipped to the users machine or is it just the development environment (asking because there's Sass present in the bundle).
> 
> As for the change from method-chaining to nesting, maybe someday in the future we'll get to use some kind of [pipe operator](https://github.com/tc39/proposal-pipeline-operator) :)
> 
> I've enjoyed looking through this and excited to learn more from you! Thanks!

We manually refactored the chaining sequences mainly to allow the plugin to work. There was some replacement of lodash functions where native functions exist (mostly for Array). Adding the plugin meant cutting the bundle down from 531.35 KB to 324.17 KB (bundle size is currently a hefty 13 MB).

File: client/app/queue/selectors.js [hunk 4]
```diff
@@ -83,88 +76,103 @@ export const appealWithDetailSelector = createSelector(
 export const getAllTasksForAppeal = createSelector(
   [getTasks, getAmaTasks, getAppealId],
   (tasks, amaTasks, appealId) => {
-    return _.filter(tasks, (task) => task.externalAppealId === appealId).
-      concat(_.filter(amaTasks, (task) => task.externalAppealId === appealId));
+    return filter(tasks, (task) => task.externalAppealId === appealId).concat(
+      filter(amaTasks, (task) => task.externalAppealId === appealId)
+    );
```
Comment (jcq) [inline client/app/queue/selectors.js:81]: I kinda feel like we should switch to vanilla JS where the lodash fns don't actually add any advantage (`Array.prototype.map`, `Array.prototype.filter`, etc, but I'm also totally fine with that being out of scope for this PR

Comment (meckila-navapbc) [reply client/app/queue/selectors.js:81]: @jcq Were there any other vanilla JS eligible functions that you saw?

Comment (jcq) [reply client/app/queue/selectors.js:81]: The most obvious were `filter` and `map` (at least when used w/ arrays); there are def more, but those were the ones that seemed like lowest-hanging fruit

Comment (meckila-navapbc) [reply client/app/queue/selectors.js:81]: What if we scoped out replacing lodash functions with the native functions in a subsequent PR? That way we can scope a PR to cleaning up unnecessary lodash functions as follow up.

Comment (jcq) [reply client/app/queue/selectors.js:81]: Yeah, that totally makes sense! I just wanted to flag as a consideration, not as a blocker for this. 😅

**Comment by charleyf** [comment]
I read through these changes as well and didn't find anything obviously wrong. I wan't able to verify the test coverage for each of these though.

**Pull request merged**